packages/launchpad/solana/client.ts

```diff
@@ -0,0 +1,483 @@
+  getFilters,
+  pk,
+  getMintAndProgram,
+  prepareTransaction,
+  signAndExecuteTransaction,
+  prepareBaseInstructions,
+} from "@streamflow/common/solana";
+import {
+  constants as streamConstants,
+  OracleType,
+  deriveTestOraclePDA,
+  deriveContractPDA,
+  deriveEscrowPDA,
+} from "@streamflow/stream/solana";
+import BN from "bn.js";
+import PQueue from "p-queue";
+
+import { LAUNCHPAD_BYTE_OFFSETS, PROGRAM_ID } from "./constants.js";
+import StreamflowLaunchpadIDL from "./descriptor/idl/streamflow_launchpad.json";
+import { StreamflowLaunchpad } from "./descriptor/streamflow_launchpad.js";
+import { deriveDepositPDA, deriveLaunchpadPDA } from "./lib/derive-accounts.js";
+import {
+  DepositAccount,
+  Launchpad,
+  IInteractSolanaExt,
+  ICreateLaunchpad,
+  IDeposit,
+  IClaimDeposits,
+  IClaimAllocatedVested,
+  IFundLaunchpad,
+} from "./types.js";
+
+interface Programs {
+  launchpadProgram: Program<StreamflowLaunchpad>;
+}
+
+interface IInitOptions {
+  clusterUrl?: string;
+  cluster?: ICluster;
+  commitment?: Commitment | ConnectionConfig;
+  programIds?: {
+    launchpad?: string;
+    vesting?: string;
+    dynamicVesting?: string;
+  };
+  sendRate?: number;
+  sendThrottler?: PQueue;
+}
+
+type CreationResult = ITransactionResult & { metadataId: PublicKey };
+
+export class SolanaLaunchpadClient {
+  connection: Connection;
+
+  cluster: ICluster;
+
+  private readonly commitment: Commitment | ConnectionConfig;
+
+  private readonly sendThrottler: PQueue;
+
+  private readonly programs: Programs;
+
+  private readonly dynamicVestingId: string;
+
+  private readonly vestingId: string;
+
+  constructor({
+    clusterUrl,
+    cluster = ICluster.Mainnet,
+    commitment = "confirmed",
+    programIds,
+    sendRate = 1,
+    sendThrottler,
+  }: IInitOptions) {
+    this.commitment = commitment;
+    if (!clusterUrl) {
+      switch (cluster) {
+        case ICluster.Mainnet:
+          clusterUrl = clusterApiUrl("mainnet-beta");
+          break;
+        case ICluster.Local:
+          clusterUrl = "http://localhost:8899";
+          break;
+        default:
+          clusterUrl = clusterApiUrl(cluster);
+          break;
+      }
+    }
+    this.connection = new Connection(clusterUrl, this.commitment);
+    this.cluster = cluster;
+    this.sendThrottler = sendThrottler ?? buildSendThrottler(sendRate);
+
+    const launchpadIdl = {
+      ...StreamflowLaunchpadIDL,
+      address: programIds?.launchpad ?? PROGRAM_ID[cluster] ?? StreamflowLaunchpadIDL.address,
+    } as StreamflowLaunchpad;
+    this.programs = {
+      launchpadProgram: new Program(launchpadIdl, {
+        connection: this.connection,
+      }) as Program<StreamflowLaunchpad>,
+    };
+    this.dynamicVestingId = programIds?.dynamicVesting
+      ? programIds.dynamicVesting
+      : streamConstants.ALIGNED_UNLOCKS_PROGRAM_ID[cluster];
+    this.vestingId = programIds?.vesting ? programIds.vesting : streamConstants.PROGRAM_ID[cluster];
```

Comment:
nit: you can make them `pk()` right away here so you don't need to cast it below